packages/lib-sourcify/src/Verification/Transformations.ts

```diff
@@ -0,0 +1,348 @@
+    const callProtectionAddress = replacedCallProtection.slice(4); // remove 0x73
+    transformations.push(CallProtectionTransformation());
+    transformationValues.callProtection = '0x' + callProtectionAddress;
+
+    return {
+      normalizedRecompiledBytecode:
+        replacedCallProtection + template.substring(callProtection.length),
+      transformations,
+      transformationValues,
+    };
+  }
+  return {
+    normalizedRecompiledBytecode: template,
+    transformations,
+    transformationValues,
+  };
+}
+
+/**
+ * Replaces the values of the immutable variables in the (onchain) deployed bytecode with zeros, so that the bytecode can be compared with the (offchain) recompiled bytecode.
+ * Easier this way because we can simply replace with zeros
+ * Example immutableReferences: {"97":[{"length":32,"start":137}],"99":[{"length":32,"start":421}]} where 97 and 99 are the AST ids
+ */
+export function extractImmutablesTransformation(
+  normalizedRecompiledBytecodeWith0x: string,
+  onchainRuntimeBytecodeWith0x: string,
+  immutableReferences: ImmutableReferences,
+  auxdataStyle: AuxdataStyle,
+) {
+  const transformations: Transformation[] = [];
+  const transformationValues: TransformationValues = {};
+  // Remove "0x" from the beginning of both bytecodes.
+  const onchainRuntimeBytecode = onchainRuntimeBytecodeWith0x.slice(2);
+  let normalizedRecompiledBytecode =
+    normalizedRecompiledBytecodeWith0x.slice(2);
+
+  Object.keys(immutableReferences).forEach((astId) => {
+    immutableReferences[astId].forEach((reference) => {
+      const { start, length } = reference;
+
+      // Save the transformation
+      transformations.push(
+        ImmutablesTransformation(
+          start,
+          astId,
+          auxdataStyle === AuxdataStyle.SOLIDITY ? 'replace' : 'insert',
+        ),
+      );
+
+      // Extract the immutable value from the onchain bytecode.
+      const immutableValue = onchainRuntimeBytecode.slice(
+        start * 2,
+        start * 2 + length * 2,
+      );
+
+      // Save the transformation value
+      if (transformationValues.immutables === undefined) {
+        transformationValues.immutables = {};
+      }
+      transformationValues.immutables[astId] = `0x${immutableValue}`;
+
+      if (auxdataStyle === AuxdataStyle.SOLIDITY) {
+        // Replace the placeholder in the recompiled bytecode with the onchain immutable value.
+        normalizedRecompiledBytecode =
+          normalizedRecompiledBytecode.slice(0, start * 2) +
+          immutableValue +
+          normalizedRecompiledBytecode.slice(start * 2 + length * 2);
+      } else if (auxdataStyle === AuxdataStyle.VYPER) {
+        // For Vyper, insert the immutable value.
+        normalizedRecompiledBytecode =
+          normalizedRecompiledBytecode + immutableValue;
+      }
+    });
+  });
+  return {
+    normalizedRecompiledBytecode: '0x' + normalizedRecompiledBytecode,
```

Comment:
Same as the comment in `extractCallProtectionTransformation`, isn't this actually a `populatedBytecode` instead of "normalized"?